File: FCTSeek-desktop/src/main/java/com/practicas/service/DepartamentoService.java
package com.practicas.service;

import com.practicas.model.Curso;
import com.practicas.model.Departamento;
import com.practicas.util.ApiClient;
import com.practicas.util.ApiClient.ApiException;

import java.util.List;

public class DepartamentoService {

    public static List<Departamento> getAll() throws ApiException {
        return ApiClient.getList("/departamentos", Departamento.class);
    }

    public static List<Curso> getCursos(long departamentoId) throws ApiException {
        return ApiClient.getList("/departamentos/" + departamentoId + "/cursos", Curso.class);
    }

    public static List<Curso> getAllCursos() throws ApiException {
        return ApiClient.getList("/cursos", Curso.class);
    }
}

File: FCTSeek-frontend-movil/src/services/departamentoService.ts
// src/services/departamentoService.ts
import { api } from '../config/api';
import { Departamento, Curso } from '../types';

export const departamentoService = {
  getAll: () => api.get<Departamento[]>('/departamentos'),

  getCursos: (departamentoId: number) =>
    api.get<Curso[]>(`/cursos/departamento/${departamentoId}`),

  getAllCursos: () => api.get<Curso[]>('/cursos'),
};
